xarray/core/utils.py

```diff
@@ -830,6 +830,26 @@ def drop_dims_from_indexers(
         )
 
 
+def dim_arg_to_dims_set(dim: Dims, all_dims: Collection) -> set:
```

Comment:
What exactly is the difference to `parse_dims` (except that it returns a tuple, which is hardly a reason to add a new method)?

Reply:
Hmm, it seems a bit more efficient to keep things as sets. For now I've added `parse_dims_as_set`, which looks like a slightly better fit for the one other use of `parse_dims` that I could find.

Were there other intended uses for `parse_dims` and `parse_ordered_dims`? I was surprised to only find one use of `parse_dims` inside Xarray.

Reply:
The intention was to ultimately use this in all methods that expect one or more dims.
But given that every method somehow handles multiple dims differently nobody was brave enough to change that because it is a somewhat breaking change.